stdlib/public/Darwin/Foundation/URL.swift

```diff
@@ -827,7 +827,8 @@ public struct URL : ReferenceConvertible, Equatable {
         } else {
             // Now we need to do something more expensive
             if var c = URLComponents(url: self, resolvingAgainstBaseURL: true) {
-                c.path = (c.path as NSString).appendingPathComponent(pathComponent)
+                let path = (c.path as NSString).appendingPathComponent(pathComponent)
+                c.path = isDirectory ? path + "/" : path
```

Comment:
Should we check if a path separator exists here at the end and not double-up?

Reply:
No need to check the existence of a separator to mimic the behavior of `NSURL.appendingPathComponent(:isDirectory:)`: it will always append a separator.